src/core/render/index.js

```diff
@@ -41,58 +43,179 @@ function formatUpdated(html, updated, fn) {
+          vueConfig,
+        ])
+        .filter(([elm, vueConfig]) => elm)
+    );
 
-        if (!isAlreadyVue) {
-          const app = window.Vue.createApp({});
+    // Template syntax, vueComponents, vueGlobalOptions
+    if (docsifyConfig.vueGlobalOptions || vueComponentNames.length) {
+      const reHasBraces = /{{2}[^{}]*}{2}/;
+      // Matches Vue full and shorthand syntax as attributes in HTML tags.
+      //
+      // Full syntax examples:
+      // v-foo, v-foo[bar], v-foo-bar, v-foo:bar-baz.prop
+      //
+      // Shorthand syntax examples:
+      // @foo, @foo.bar, @foo.bar.baz, @[foo], :foo, :[foo]
+      //
+      // Markup examples:
+      // <div v-html>{{ html }}</div>
+      // <div v-text="msg"></div>
+      // <div v-bind:text-content.prop="text">
+      // <button v-on:click="doThis"></button>
+      // <button v-on:click.once="doThis"></button>
+      // <button v-on:[event]="doThis"></button>
+      // <button @click.stop.prevent="doThis">
+      // <a :href="url">
+      // <a :[key]="url">
+      const reHasDirective = /<[^>/]+\s([@:]|v-)[\w-:.[\]]+[=>\s]/;
+
+      vueMountData.push(
+        ...dom
+          .findAll('.markdown-section > *')
+          // Remove duplicates
+          .filter(elm => !vueMountData.some(([e, c]) => e === elm))
+          // Detect Vue content
+          .filter(elm => {
+            const isVueMount =
+              // is a component
+              elm.tagName.toLowerCase() in
+                (docsifyConfig.vueComponents || {}) ||
+              // has a component(s)
+              elm.querySelector(vueComponentNames.join(',') || null) ||
+              // has curly braces
+              reHasBraces.test(elm.outerHTML) ||
+              // has content directive
+              reHasDirective.test(elm.outerHTML);
+
+            return isVueMount;
+          })
+          .map(elm => {
+            // Clone global configuration
+            const vueConfig = Object.assign(
+              {},
+              docsifyConfig.vueGlobalOptions || {}
+            );
+
+            // Replace vueGlobalOptions data() return value with shared data object.
+            // This provides a global store for all Vue instances that receive
+            // vueGlobalOptions as their configuration.
+            if (vueGlobalData) {
+              vueConfig.data = function() {
+                return vueGlobalData;
+              };
+            }
+
+            return [elm, vueConfig];
+          })
+      );
+    }
 
-          app.mount(elm);
-        }
-      }
-      // Vue 2
-      else if (vueVersion === 2) {
-        const isAlreadyVue = Boolean(elm.__vue__ && elm.__vue__._isVue);
-
-        if (!isAlreadyVue) {
-          new window.Vue({
-            mounted: function() {
-              this.$destroy();
-            },
-          }).$mount(elm);
+    // Mount
+    for (const [mountElm, vueConfig] of vueMountData) {
+      const isVueAttr = 'data-isvue';
```

Comment:
`data-isvue` isn't documented in `vue.md`

Reply:
Oh, I see, it's a "private" attribute. What about using a more private thing like a JS property like `__docsify_is_vue`?

Reply:
I'm using a data attribute because it is faster and easier to detect Vue instances mounted by docsify using `querySelector('data-issue')` than it is to manually loop through a potentially large collection of nodes and test for a JS property like `__docsifyIsVue`. The decision was made easier once I noticed that Vue 3 adds its own data attribute to all Vue instances: `data-v-app`.